test/testResourceNoTranslation.js

```diff
@@ -0,0 +1,353 @@
+    },
+
+    testResourceNoTranslationDNT: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceNoTranslation();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "translation.test",
+                sourceLocale: "en-US",
+                source: 'This is the source string.',
+                dnt: true,
+                targetLocale: "en-GB",
+                target: "This is the source string.",
+                pathName: "a/b/c.xliff",
+                state: "new"
+            }),
+            file: "x/y"
+        });
+
+        // no results because the resource has the Do Not
+        // Translate flag (dnt)
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceNoTranslationKababText: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceNoTranslation();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "translation.test",
+                sourceLocale: "en-US",
+                source: 'this-is-kabab-text',
+                targetLocale: "de-DE",
+                target: "this-is-kabab-text",
+                pathName: "a/b/c.xliff",
+                state: "new"
+            }),
+            file: "x/y"
+        });
+
+        // no results because kabab text is automatically DNT
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceNoTranslationCamelText: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceNoTranslation();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "translation.test",
+                sourceLocale: "en-US",
+                source: 'thisIsCamelCaseText',
+                targetLocale: "de-DE",
+                target: "thisIsCamelCaseText",
+                pathName: "a/b/c.xliff",
+                state: "new"
+            }),
+            file: "x/y"
+        });
+
+        // no results because camel case text is automatically DNT
+        test.ok(!actual);
+
+        test.done();
+    },
+
+    testResourceNoTranslationSnakeText: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceNoTranslation();
+        test.ok(rule);
+
+        const actual = rule.match({
+            locale: "de-DE",
+            resource: new ResourceString({
+                key: "translation.test",
+                sourceLocale: "en-US",
+                source: 'this_is_snake_text',
+                targetLocale: "de-DE",
+                target: "this_is_snake_text",
+                pathName: "a/b/c.xliff",
+                state: "new"
+            }),
+            file: "x/y"
+        });
+
+        // no results because snake text is automatically DNT
```

Comment:
Remove case-related tests (text being x-case is not the reason for DNT)

Reply:
updated